crates/protogen/src/sqlexec/logical_plan.rs

```diff
@@ -0,0 +1,41 @@
+use datafusion_proto::protobuf::{DfSchema, LogicalPlanNode, OwnedTableReference};
+use prost::{Message, Oneof};
+
+#[derive(Clone, PartialEq, Message)]
+pub struct CreateTable {
+    #[prost(message, optional, tag = "1")]
+    pub table_name: Option<OwnedTableReference>,
+    #[prost(bool, tag = "2")]
+    pub if_not_exists: bool,
+    #[prost(message, optional, tag = "3")]
+    pub schema: Option<DfSchema>,
+    #[prost(message, optional, tag = "4")]
+    pub source: Option<LogicalPlanNode>,
+}
+
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, Message)]
+pub struct LogicalPlanExtension {
```

Comment:
since protobuf messages are not self-describing, we have to roll all extensions up into a single message containing a `oneof`. Otherwise we lose context of what the original type was & have to try all variants until we find a match.